modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionManager.java

```diff
@@ -146,21 +148,31 @@ protected void setExtendedProperties(
 			return;
 		}
 
-		Map<String, Serializable> extendedProperties = new HashMap<>();
+		for (ExtensionProvider extensionProvider : extensionProviders) {
+			Map<String, PropertyDefinition> propertyDefinitions =
+				extensionProvider.getExtendedPropertyDefinitions(
+					user.getCompanyId(), className);
 
-		for (Map.Entry<String, Object> entry : values.entrySet()) {
-			extendedProperties.put(
-				entry.getKey(), (Serializable)entry.getValue());
-		}
+			Map<String, Serializable> extendedProperties = new HashMap<>();
 
-		for (ExtensionProvider extensionProvider : extensionProviders) {
-			try {
-				extensionProvider.setExtendedProperties(
-					user.getCompanyId(), user.getUserId(), className, entity,
-					extendedProperties);
+			for (Map.Entry<String, Object> entry : values.entrySet()) {
+				String propertyKey = entry.getKey();
+
+				if (propertyDefinitions.containsKey(propertyKey)) {
+					extendedProperties.put(
+						propertyKey, (Serializable)entry.getValue());
+				}
 			}
-			catch (Exception exception) {
-				_log.error(exception);
+
+			if (MapUtil.isNotEmpty(extendedProperties)) {
+				try {
+					extensionProvider.setExtendedProperties(
+						user.getCompanyId(), user.getUserId(), className,
+						entity, extendedProperties);
+				}
+				catch (Exception exception) {
+					_log.error(exception);
```

Comment:
Not sure if we should log and ignore the error or throws the exception instead.

Why is that code there? What kind of errors could happen?

Reply:
As it was explained by Objects team, this code could be removed